### DISCUSSION TITLE
Get commit ids/messages of a file

### DISCUSSION THREAD

**Discussion by zitaszocs** [libgit2/libgit2#6081] [Q&A]
Hi all,

There is a possibility to get commit ids by path?
I have seen that the libgit2sharp version has a list of commits in the Repository object which can be queried by a file path, but I am unable to find something similar here.

**Comment by extrawurst** [comment]
libgit2sharp does not use anything that libgit2 does not provide - it builds on top of it - so it should be easy to achieve in C++ too. Can you point me to the place in libgit2sharp you refer to?

**Reply by zitaszocs** [reply]
I am looking for something similar to git command: 'git log -n 1 --pretty=format:"%H %s" -- filename' or 'git log --pretty=format:"%H %s" -- filename' and I found the following solution for libgit2sharp:
        Repository gitRepository = new Repository("myrepo");
            var logs = gitRepository.Commits.QueryBy(@"my_file_path");
            foreach (var log in logs)
            {
                System.Console.WriteLine(log.Commit.Message);     
            }

**Reply by lhchavez** [reply]
We currently don't have a single direct API for doing that, _but_ there's an issue to add one such API: https://github.com/libgit2/libgit2/issues/5758 (and to make it _fast_ with the aid of git 2.29's bloom filters, if the repository supports it).

Having said that, this is totally achievable with the current API! One way of achieving this is to traverse the commit log using the [revwalk API](https://libgit2.org/libgit2/ex/HEAD/log.html). for every commit, a call to [`git_tree_entry_bypath`](https://libgit2.org/libgit2/#HEAD/group/tree/git_tree_entry_bypath) can tell whether the referenced path was modified when compared against its parent(s).

**Reply by zitaszocs** [reply]
Thank you, I will try it out.
